packages/abi-coder/src/abi-coder.ts

```diff
@@ -27,23 +24,100 @@ import {
   tupleRegEx,
   OPTION_CODER_TYPE,
   VEC_CODER_TYPE,
+  genericRegEx,
 } from './constants';
-import type { JsonAbiFragmentType } from './json-abi';
-import { filterEmptyParams, getVectorAdjustments, hasOptionTypes } from './utilities';
+import type { JsonAbi, JsonAbiArgument, JsonAbiType } from './json-abi';
 
 const logger = new Logger(versions.FUELS);
 
-export default class AbiCoder {
-  constructor() {
-    logger.checkNew(new.target, AbiCoder);
+export abstract class AbiCoder {
+  private static getImplicitGenericTypeParameters(
```

Comment:
Implicitly generic type parameters happen when a generic type parameter gets passed to e.g. an array in sway.
```
struct MyGenericStruct<A> {
 arr: [A; 5]
}
```
In the json abi, this gets encoded as an array of `generic A`, but there are no `typeParameters` in this generated array abi type, so it's _implicitly_ generic with `generic A`, because the presence of `typeParameters` indicates that the abi type is _explicitly_ generic.